### DISCUSSION TITLE
Show gpx track on map

### DISCUSSION THREAD

**Discussion by ramGHnet** [javalent/obsidian-leaflet#165] [Ideas]
I would like to display recorded gpx tracks (bicycle, car, camper, etc) on an obsidian-leafleat map. 

With leaflet I used the [leaflet-gpx plugin](https://github.com/mpetazzoni/leaflet-gpx) for this.

Could you implement the feature in obsidian-leaflet as well?

**Comment by valentine195** [comment]
This would be handled by the plugin similarly to GeoJSON, I think (as in, a parameter in the source block). Does that work for you?

**Reply by ramGHnet** [reply]
geojson is also ok. Then I convert my 500 gpx files with python to geojson, generate an obsidian note (with obsidian-leaflet content) and see what happens.

**Comment by Moonbase59** [comment]
You could also convert your GPX to GeoJSON using one of the following services:

- <https://mapbox.github.io/togeojson/> (copy-paste)
- <https://products.aspose.app/gis/conversion/gpx-to-geojson> (drop file)

That would allow using the existing GeoJSON features, plus editing the file in JSON. (That’s what I do with my _MyTrails_ data.)

Then again, modern GPX tracker data contains a lot more niceties, like, for instance

- elevation
- speed
- distance
- moving pace/speed
- heartrate
- temperature
- … and more.

**Comment by Moonbase59** [comment]
Thinking about it, it _might_ be nice to have direct GPX support in Obsidian Leaflet: I find myself using [MyTrails (Android)](https://play.google.com/store/apps/details?id=com.frogsparks.mytrails&referrer=frogsparks) ever more often, and keeping separate GPX and GeoJSON versions isn’t _so_ nice. Besides, I have to convert every time, losing data if doing GPX → GeoJSON → GPX (I like to transfer the edited GPX _back_ to the smartphone).

GPX has for me the advantage that:

- I can sync the GPX data from my mobile with my laptop/desktop
- I can edit the GPX on my laptop/desktop
- I can store waypoints (POIs)
- GPX has much more data, like _travel direction_, speed, track lengths, height difference, timestamps etc.
- I can (manually) use the stored waypoint data for my Obidian Leaflet notes

MyTrails (and other GPX apps) show tracks and statistics quite nicely, might be nice to have that in Obsidian Leaflet too, without converting …

![Screenshot_20210730-222329](https://user-images.githubusercontent.com/3706922/127710004-43659880-4fe7-42a0-abfa-d55b01189cd6.png)
![Screenshot_20210730-222420](https://user-images.githubusercontent.com/3706922/127710027-1a9cab19-c668-48ea-a9d3-05ae7299dabd.png)

**Comment by valentine195** [comment]
Basic gpx support added in 3.24.0. See [here](https://github.com/valentine195/obsidian-leaflet-plugin#gpx).

More features will be coming, but for now you can display them.
